discovery/gossiper.go

```diff
@@ -2321,16 +2324,10 @@ func (d *AuthenticatedGossiper) isMsgStale(msg lnwire.Message) bool {
 		return chanInfo.AuthProof != nil
 
 	case *lnwire.ChannelUpdate:
-		// The MessageStore will always store the latest ChannelUpdate
-		// as it is not aware of its timestamp (by design), so it will
-		// never be stale. We should still however check if the channel
-		// is part of our graph. If it's not, we can mark it as stale.
-		_, _, _, err := d.cfg.Router.GetChannelByID(msg.ShortChannelID)
-		if err != nil && err != channeldb.ErrEdgeNotFound {
-			log.Debugf("Unable to retrieve channel=%v from graph: "+
-				"%v", err)
-		}
-		return err == channeldb.ErrEdgeNotFound
+		timestamp := time.Unix(int64(msg.Timestamp), 0)
+		return d.cfg.Router.IsStaleEdgePolicy(
```

Comment:
are we sure we will never overwrite the policy with a new one before we have been able to send it to the peer?

Reply:
Even if we overwrite it, then it means we have a newer version which we should be sending. In this case, we'll only reliably send if the channel is still unannounced, because we assume any peer can receive the message as long as the channel is announced.

Reply:
The scenario I'm thinking about is that the channel stay unannounced (private?) and we for some reason create a new edge policy that is added to the graph (but not to the message store!). In this scenario the policy found in the message store is considered stale.

We need to make sure we either add the updated policy to the message store in this case, or keep the update around until the channel has been announced.

Reply:
Not sure I follow. If the edge for the policy that we're applying is still unannounced, then we'll also add it to the message store. See:
https://github.com/lightningnetwork/lnd/blob/e3c65e1a4935dbf9ba91cb326ea78b5161748dad/discovery/gossiper.go#L1961

Reply:
Resolved offline.